protocol/messenger_linkpreview_test.go

```diff
@@ -378,3 +370,53 @@ func (s *MessengerLinkPreviewsTestSuite) Test_UnfurlURLs_Image() {
 	s.Require().Equal(expected.Thumbnail.URL, preview.Thumbnail.URL)
 	s.assertContainsLongString(expected.Thumbnail.DataURI, preview.Thumbnail.DataURI, 100)
 }
+
+func (s *MessengerLinkPreviewsTestSuite) Test_UnfurlURLs_Settings() {
+	u := "https://github.com"
+	requestsCount := 0
+
+	transport := StubTransport{}
+	transport.AddURLMatcherRoundTrip(
+		u,
+		func(req *http.Request) *http.Response {
+			requestsCount++
+			responseBody := []byte(`<html><head><meta property="og:title" content="TestTitle"></head></html>`)
+			return &http.Response{
+				StatusCode: http.StatusOK,
+				Body:       ioutil.NopCloser(bytes.NewBuffer(responseBody)),
+			}
+		},
+	)
+	stubbedClient := http.Client{Transport: &transport}
+
+	// Test `AlwaysAsk`
+	// NOTE: on status-go side `AlwaysAsk` == `EnableAll`, "asking" should be processed by the app
+
+	requestsCount = 0
+	err := s.m.settings.SaveSettingField(settings.URLUnfurlingMode, settings.URLUnfurlingAlwaysAsk)
+	s.Require().NoError(err)
+
+	linkPreviews, err := s.m.UnfurlURLs(&stubbedClient, []string{u})
+	s.Require().NoError(err)
+	s.Require().Len(linkPreviews, 1)
+	s.Require().Equal(requestsCount, 1)
+
+	// Test `EnableAll`
+	requestsCount = 0
+	err = s.m.settings.SaveSettingField(settings.URLUnfurlingMode, settings.URLUnfurlingEnableAll)
+	s.Require().NoError(err)
+
+	linkPreviews, err = s.m.UnfurlURLs(&stubbedClient, []string{u})
+	s.Require().NoError(err)
+	s.Require().Len(linkPreviews, 1)
+
+	// Test `DisableAll`
+	requestsCount = 0
```

Comment:
The `requestsCount` is being reset, but we're not asserting anything about it. I guess we want to do the same as `s.Require().Equal(requestsCount, 1)`?